modules/gdscript/gdscript_parser.cpp

```diff
@@ -3261,31 +3261,85 @@ GDScriptParser::TypeNode *GDScriptParser::parse_type(bool p_allow_void) {
 }
 
 #ifdef TOOLS_ENABLED
-static bool _in_codeblock(String p_line, bool p_already_in, int *r_block_begins = nullptr) {
-	int start_block = p_line.rfind("[codeblock]");
-	int end_block = p_line.rfind("[/codeblock]");
-
-	if (start_block != -1 && r_block_begins) {
-		*r_block_begins = start_block;
+static String _process_doc_line(const String &p_line, const String &p_text, bool *r_in_codeblock) {
+	ERR_FAIL_COND_V(r_in_codeblock == nullptr, String());
+
+	String line = p_line;
+	if (*r_in_codeblock) {
+		int i = 0;
+		for (; i < line.length(); i++) {
+			if (line[i] != ' ') {
+				break;
+			}
+		}
+		// ## [codeblock]
+		// ## if true:
+		// ##^
+		// ##     print(42)
+		// ##^^^^^
+		line = line.substr(i % 4);
```

Comment:
I'm not sure how best to handle this.

Reply:
Why not use lstrip?

Reply:
We need to keep the indentation inside the codeblock. Ideally, all of these options

```gdscript
##[codeblock]
##if true:
##    print(42)
```

```gdscript
## [codeblock]
## if true:
##     print(42)
```

```gdscript
##  [codeblock]
##  if true:
##      print(42)
```

should be converted to

```
[codeblock]
if true:
    print(42)
```

Reply:
Done.